### PR TITLE
Job posting common item

### PR DESCRIPTION
**Pull request by kishan3**
Doc: https://docs.google.com/spreadsheets/d/1cRZ02Aw9M3d2GF3KLL0Ju0aS82y9WU9H3AtyLVkqywE/edit#gid=0
Unified schema is merged for this.

### CODE REVIEW

**Comment by kishan3** [comment]
@wRAR @Gallaecio can you please check this when you time? thanks.

**Comment by wRAR** [comment]
It also needs running pre-commit.

**Comment by Gallaecio** [comment]
See https://docs.scrapy.org/en/latest/contributing.html#pre-commit

To run it on all files now, to make up for not having it installed while you made those commits, run `pre-commit run --all`.

**Comment by kishan3** [comment]
> See https://docs.scrapy.org/en/latest/contributing.html#pre-commit
> 
> To run it on all files now, to make up for not having it installed while you made those commits, run `pre-commit run --all`.

Thanks @Gallaecio done. btw all tox checks fails I am not sure why.

**Comment by wRAR** [comment]
The test_metadata failure: you need to add the new class to `METADATA_FIELDS`.

The test_page_pairs failure: I assume it's because BaseJobPostingPage needs to be imported into the top-level `__init__.py`.

File: docs/reference/pages.rst [hunk 1]
```diff
@@ -80,6 +80,10 @@ Real estate
 .. autoclass:: zyte_common_items.RealEstatePage(**kwargs)
    :show-inheritance:
 
+Job posting
+===========
+.. autoclass:: zyte_common_items.JobPostingPage(**kwargs)
```
Comment (wRAR) [inline docs/reference/pages.rst:85]: BaseJobPostingPage needs to be in this doc too.

File: zyte_common_items/components.py [hunk 2]
```diff
@@ -447,3 +447,55 @@ class ProbabilityRequest(Request):
...
+
+    #: Organization information as available on the website.
+    nameRaw: Optional[str] = None
+
+    #: Identifier of the organization used by job posting website.
+    id: Optional[str] = None
+
+
+@attrs.define(kw_only=True)
+class JobPostingMetadata(Metadata):
```
Comment (wRAR) [inline zyte_common_items/components.py:498]: Can you please move this to other metadata classes? I think they are in alphabetical order there.

File: zyte_common_items/pages.py [hunk 3]
```diff
@@ -206,3 +208,13 @@ class ProductNavigationPage(
 class RealEstatePage(Page, Returns[RealEstate], HasMetadata[RealEstateMetadata]):
     class Processors(Page.Processors):
         breadcrumbs = [breadcrumbs_processor]
+
+
+class BaseJobPostingPage(
```
Comment (wRAR) [inline zyte_common_items/pages.py:213]: Can you please move this to other Base*Page classes?

File: zyte_common_items/pages.py [hunk 4]
```diff
@@ -206,3 +208,13 @@ class ProductNavigationPage(
...
+    BasePage, Returns[JobPosting], HasMetadata[JobPostingMetadata]
+):
+    pass
+
+
+class JobPostingPage(Page, Returns[JobPosting], HasMetadata[JobPostingMetadata]):
```
Comment (wRAR) [inline zyte_common_items/pages.py:219]: I think this should go after BusinessPlacePage to keep the list sorted.

**Comment by Gallaecio** [comment]
For the record, if you run `pre-commit install`, it will run automatically when you commit, no need to run `pre-commit run` manually.

File: docs/reference/components.rst [hunk 5]
```diff
@@ -101,3 +101,15 @@ Item metadata components
...
+   :members: dateDownloaded, probability
+
+.. autoclass:: zyte_common_items.BaseSalary(**kwargs)
+   :members: raw, valueMin, valueMax, rateType, currency, currencyRaw
+
+.. autoclass:: zyte_common_items.HiringOrganization(**kwargs)
+   :members: name, nameRaw, id
+
+.. autoclass:: zyte_common_items.JobLocation(**kwargs)
+   :members: raw
```
Comment (Gallaecio) [inline docs/reference/components.rst:115]: The metadata class should stay in this section, but properly sorted, alphabetically.

The rest of components should be on the earlier section.

**Review by Gallaecio — APPROVED** [review]

**Comment by kishan3** [comment]
> For the record, if you run `pre-commit install` once, it will later run automatically when you commit, no need to run `pre-commit run` manually.

Ok thanks when I pushed https://github.com/zytedata/zyte-common-items/pull/48/commits/d18fcfa4cd8060a4d913d398bdcb9bc92b205f07 it failed so I pushed manually. Not sure why it didn't work I have pre-commit installed.

**Comment by wRAR** [comment]
Thanks!

**Comment by kishan3** [comment]
Thank you guys for the comprehensive review and support on this.

**Pull request merged**